app/src/main/kotlin/com/wire/android/ui/authentication/welcome/Welcome.kt

```diff
@@ -80,50 +82,64 @@ private fun WelcomeContent(navController: NavController) {
                 WelcomeCarousel()
             }
 
-            Column(modifier = Modifier.padding(top = 40.dp, bottom = 52.dp, start = 16.dp, end = 16.dp),) {
+            Column(modifier = Modifier.padding(
+                vertical = MaterialTheme.wireDimensions.welcomeVerticalSpacing,
+                horizontal = MaterialTheme.wireDimensions.welcomeButtonHorizontalPadding
+            )) {
                 LoginButton {
                     navController.navigate(AuthDestination.loginScreen)
                 }
                 CreateEnterpriseAccountButton {
                     navController.navigate((AuthDestination.createEnterpriseAccount))
-
                 }
             }
 
-            WelcomeFooter(modifier = Modifier.padding(bottom = 56.dp, start = 16.dp, end = 16.dp),
+            WelcomeFooter(modifier = Modifier
+                .padding(horizontal = MaterialTheme.wireDimensions.welcomeTextHorizontalPadding),
                 onPrivateAccountClick = {
                     navController.navigate(AuthDestination.createPrivateAccountScreen)
                 })
         }
     }
 }
 
-@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class)
+@OptIn(ExperimentalPagerApi::class, ExperimentalFoundationApi::class, ExperimentalCoroutinesApi::class)
 @Composable
 private fun WelcomeCarousel() {
     val delay = integerResource(id = R.integer.welcome_carousel_item_time_ms)
     val icons: List<Int> = typedArrayResource(id = R.array.welcome_carousel_icons).drawableResIdList()
     val texts: List<String> = stringArrayResource(id = R.array.welcome_carousel_texts).toList()
     val items: List<Pair<Int, String>> = icons zip texts
-    val circularItemsList = listOf<Pair<Int, String>>().plus(items.last()).plus(items)
-    val pageState = rememberPagerState(initialPage = 1)
-
-    LaunchedEffect(pageState.currentPage) {
-        if (pageState.currentPage == circularItemsList.lastIndex) {
-            pageState.scrollToPage(0)
-        } else {
-            delay(delay.toLong())
-            pageState.animateScrollToPage(pageState.currentPage + 1)
-        }
+    val circularItemsList = listOf<Pair<Int, String>>().plus(items.last()).plus(items).plus(items.first())
+    val initialPage = 1
+    val pageState = rememberPagerState(initialPage = initialPage)
+
+    LaunchedEffect(pageState) {
+        snapshotFlow { pageState.currentPage }
+            .distinctUntilChanged()
+            .scan(initialPage to initialPage) { (_, previousPage), currentPage -> previousPage to currentPage }
+            .flatMapLatest { (previousPage, currentPage) ->
+                when {
+                    shouldJumpToStart(previousPage, currentPage, circularItemsList.lastIndex, initialPage) ->
+                        flow { emit(initialPage to false) }
```

Comment:
abit unclear to me when reading this line what you are emitting and why, later on in the collect it becomes a bit more clear though

maybe instead of using a Pair object, create a custom data class like : 

data ScrollState class(val page: Int, val is animating : Boolean) (dunno if i get the names right :P ) 

and later emit it with named parameters like emit(ScrollState(page = initialPage, isAnimating = false)